lib/srv/alpnproxy/local_proxy_test.go

```diff
@@ -269,3 +413,20 @@ func createAWSAccessProxySuite(t *testing.T, cred *credentials.Credentials) *Loc
 	}()
 	return lp
 }
+
+func requireExpiredCertErr(t require.TestingT, err error, _ ...interface{}) {
```

Comment:
I would add `t.Helper()` to these.

Reply:
these functions match the type signature for `require.ErrorAssertFunc` which takes a `require.TestingT` instead of `*testing.T`, and `require.testingT` does not have a `Helper` method. I can use a type assertion like `testify/require` does for all of its assertion functions, but I see no difference in the stack trace produced from doing that:

```
func requireCertSubjectUserErr(t require.TestingT, err error, _ ...interface{}) {
	if h, ok := t.(*testing.T); ok {
		h.Helper()
	}
	if !assert.Error(t, err) {
		t.FailNow()
	}
	if !assert.ErrorContains(t, err, "certificate subject is for user") {
		t.FailNow()
	}
}
```
stack trace when failing:
```
    local_proxy_test.go:388: 
        	Error Trace:	/Users/gavin/work/teleport/lib/srv/alpnproxy/local_proxy_test.go:431
        	           				/Users/gavin/work/teleport/lib/srv/alpnproxy/local_proxy_test.go:388
        	Error:      	Error "x509: certificate has expired or is not yet valid: current time 2022-11-09T22:34:12Z is before 2022-11-09T23:33:12Z" does not contain "certificate subject is for user"
        	Test:       	TestCheckDBCerts/detects_clock_is_after_cert_expires
```

stack trace when failing with the definition you see in the PR:
```
    local_proxy_test.go:426: 
        	Error Trace:	/Users/gavin/work/teleport/lib/srv/alpnproxy/local_proxy_test.go:426
        	           				/Users/gavin/work/teleport/lib/srv/alpnproxy/local_proxy_test.go:388
        	Error:      	Error "x509: certificate has expired or is not yet valid: current time 2022-11-09T22:35:24Z is before 2022-11-09T23:34:24Z" does not contain "certificate subject is for user"
        	Test:       	TestCheckDBCerts/detects_clock_is_after_cert_expires
```

Reply:
Oh, it changed the first line. Is that all that does?

Reply:
Yeah, the idea is that you want the first line number you see in the failure message to be in a specific test and not in a helper method that several different tests use. Makes it easier to track down the failure.

That said, I didn't realize you'd have to do the type assertion, that kind of stinks :-( I'm okay with this way or the original way, your call.

Reply:
Might as well keep it, testify does basically the same thing everywhere